fix(mev_simBundle): Return fuller log objects.

ryanschneider opened the pull request:
DRAFT since it uses an unpublished alloy fork in 65a46fef2720d9abf8a93cd036714a386bc217f0 since it needs https://github.com/alloy-rs/alloy/pull/2061.

This gets a closer to being a drop-in replacement for mev-geth, but is still missing `blockHash` and `blockNumber` values in the `txLogs` objects.

I think the cleanest way to get those would actually be to reuse some of the helpers from the `eth_simulateV1` `simulate` namespace, namely `simulate::build_simulated_block(...)` but maybe a couple more.

mattsse commented on crates/rpc/rpc/src/eth/sim_bundle.rs:
```diff
@@ -335,7 +336,21 @@ where
+                                block_timestamp: None,
+                                transaction_hash: Some(*item.tx.tx_hash()),
+                                transaction_index: Some(tx_index as u64),
+                                log_index: Some(log_index as u64),
```
isn't this the index in the block? so we should keep a counter across the tx_index loop?

ryanschneider replied:
Yup good call, fixed in https://github.com/paradigmxyz/reth/pull/14506/commits/be44fa823b592bd36d16bc48a13b77fedb57c651

mattsse commented on crates/rpc/rpc/src/eth/sim_bundle.rs:
```diff
@@ -335,7 +336,21 @@ where
+                            .enumerate()
+                            .map(|(log_index, log)| alloy_rpc_types_eth::Log {
+                                inner: log.clone(),
+                                block_hash: None,
+                                block_number: None,
+                                block_timestamp: None,
```
should those even be set?

I guess block_number and timestamp could be derived from the parent block, but maybe those should all just be None

ryanschneider replied:
I dug into how the original mev-geth worked, and it actually does something very similiar to the reth implementation of `eth_simulateV1` where it builds a block on top of the chain w/ all the txs in it, and the logs contain the computed blockHash and number for that.  So I think that would be the cleaner longer term fix, but for now its not really necessary.  The txHash & index are the main things that are used to lookup logs in the response, the block number/hash isn't really useful (the number can obviously be derived and the block hash isn't meaningful in this context).

mattsse reviewed (commented):
ty, one q

ryanschneider commented:
@mattsse I rebased against main and squashed everything into two commits: https://github.com/paradigmxyz/reth/pull/14506/commits/a303e83532e19168d96d0d75c470592b3c529942 which could be cherry-picked when this gets unblocked by a new alloy release, and https://github.com/paradigmxyz/reth/pull/14506/commits/c94101ead6b59a0253136d5a5d500e7dd6e0f9bd which should be dropped since it's just my `Cargo.toml` patch.

mattsse commented:
included in #14891

ryanschneider commented:
Thanks!
